lib/fs/walk.go

```diff
@@ -0,0 +1,113 @@
+package fs
+
+import (
+	"os"
+	"path/filepath"
+
+	"polydawn.net/repeatr/lib/treewalk"
+)
+
+type WalkFunc func(filenode *FilewalkNode) error
+
+/*
+	Walks a filesystem.
+
+	This is much like the standard library's `path/filepath.Walk`, except
+	it's based on `treewalk`, which means it supports both pre- and post-order
+	traversals.
+
+	All paths begin in `.`, e.g. you'll see a series like	`{".", "./a", "./a/b"}`, etc.
+	This is identical in behavior to `filepath.Clean` having been called on each path.
+
+	Symlinks are not followed.
+
+	The traversal order of siblings is *not* guaranteed, and is *not* necessarily
+	stable.
```

Comment:
Posix filesystems aren't required to give you things in sorted order.  (Most of the golang readdirs functions do the sort in-application.)

Since we tend to have to do a big-picture sort any time we care, and at other times don't care at all, it seemed easy to leave it up unsorted here.

Can add a bool flag for sorting if we hit a need for it.